### PR TITLE
Update resource to require token for metric endpoint

### PR DESCRIPTION
**Pull request by ZacharyWills**
Updated instance resource to add token requirements for AWS Instance.

in reference to issue #51

### CODE REVIEW

**Comment by mwengren** [comment]
@ZacharyWills There are some merge conflicts with this currently.  Would you mind rebasing and resubmitting after #54 can be merged, hopefully?

**Comment by ZacharyWills** [comment]
@mwengren Give it another look and let me know if it's ready to merge.

**Comment by mwengren** [comment]
@ZacharyWills My preference would be to keep merge commits out of PRs, if possible.  It will make for a cleaner project history.  

I think you could reapply your changes in the original three commits by running something like

```
git rebase -i main
```

from your feature branch to rewind and then reapply them from the head of the `main` branch.  You might even be able to skip the `-i` flag in this case and apply them directly without editing the commit history interactively.

Then just do a force push to the same branch on your remote and it should clean things up a bit in this PR.  

I don't see any issues with the changes, but to be honest, I don't really have the bandwidth to test them so we'll have to test on the next time we redeploy.

If @patrick-tripp or @jonmjoyce don't see any issues, we can merge this once you've updated the branch I think.

File: terraform/main.tf [hunk 1]
```diff
@@ -351,24 +347,33 @@ resource "aws_instance" "head_node" {
...
-  user_data = data.template_file.init_instance.rendered
+  user_data            = templatefile("init_template.tpl", { efs_name = aws_efs_file_system.main_efs.dns_name, ami_name = "${var.name_tag}-${random_pet.ami_id.id}", aws_region = var.preferred_region, project = var.project_tag })
 
   # associate_public_ip_address = true
   network_interface {
+    device_index         = 0 # MUST be 0
+    network_interface_id = var.use_efa == true ? aws_network_interface.efa_network_adapter.id : aws_network_interface.standard.id
     device_index = 0    # MUST be 0
     network_interface_id = aws_network_interface.head_node.id
   }
```
Comment (jonmjoyce) [inline terraform/main.tf:379]: This will reassign network_interface_id; should line 377/378 be deleted?

**Review by jonmjoyce — COMMENTED** [review]
Most of these changes are changes in spacing/formatting so it's difficult to see what the actual code changes were. But the few I found looked fine without testing.

File: terraform/main.tf (hunk 1 above)
Comment (mwengren) [reply terraform/main.tf:379]: Looks like this was introduced in the most recent merge commit af85bfaf9e976a30ae35d341b690a909f1d896ff.  Probably the two added lines should just be removed.

Comment (patrick-tripp) [reply terraform/main.tf:379]: Yes, remove the two added lines (those are the from the previous version). EFA isn't needed on the head node.

**Review by patrick-tripp — APPROVED** [review]

**Review by patrick-tripp — CHANGES_REQUESTED** [review]
Remove the two added lines in main.tf as discussed.

File: terraform/main.tf (hunk 1 above)
Comment (ZacharyWills) [reply terraform/main.tf:384]: Removed these lines, not sure if we still needed the `device_index` line at all.

Take a look.

**Review by patrick-tripp — APPROVED** [review]

File: terraform/main.tf [hunk 2]
```diff
@@ -351,33 +348,46 @@ resource "aws_instance" "head_node" {
...
+  }
+  instance_type        = var.instance_type
+
+=======
+  # Can optionally use redhat - use the parameterized
+  # ami = data.aws_ami.rh_ufs.id
+  metadata_options {
+	 http_tokens = "required"
+  }	
+>>>>>>> 7185af3 (Update resource to require token for metric endpoint)
```
[the remark this answers is not in the document]
Comment (patrick-tripp) [reply terraform/main.tf:364]: Thanks Felipe. It should be ok now.

[the remark this answers is not in the document]
Comment (ocefpaf) [reply terraform/main.tf:364]: Fixed in https://github.com/ioos/Cloud-Sandbox/pull/61

**Pull request merged**